state.go

```diff
@@ -257,7 +258,31 @@ func (m *Memberlist) probeNode(node *nodeState) {
 		}
 	}
 
-	// Wait for the acks or timeout
+	// Also make an attempt to contact the node directly over TCP. This
+	// helps prevent confused clients who get isolated from UDP traffic
+	// but can still speak TCP (which also means they can possibly report
+	// misinformation to other nodes via anti-entropy), avoiding flapping in
+	// the cluster.
+	fallbackCh := make(chan bool)
+	if node.PMax >= 3 {
```

Comment:
We should also check if our own protocol version is at least 3. e.g. both sides must be >= 3